home/migrations/0099_migrate_content_page_templates_to_standalone_templates.py

```diff
@@ -1,37 +1,102 @@
+    duplicate_whatsapp_titles = (
+        ContentPage.objects.filter(is_whatsapp_template=True)
+        .values("whatsapp_title")
+        .annotate(count=Count("whatsapp_title"))
+        .filter(count__gt=1)
+        .values_list("whatsapp_title", flat=True)
+    )
+    for whatsapp_title in duplicate_whatsapp_titles:
+        pages = ContentPage.objects.filter(whatsapp_title=whatsapp_title)
+        while pages.count() > 1:
+            page = pages.first()
+            suffix = 1
+            candidate_whatsapp_title = whatsapp_title
+            while ContentPage.objects.filter(
+                whatsapp_title=candidate_whatsapp_title
+            ).exists():
+                suffix += 1
+                candidate_whatsapp_title = f"{whatsapp_title}-{suffix}"
+            page.whatsapp_title = candidate_whatsapp_title
+            page.save(update_fields=["whatsapp_title"])
+
+
+def rename_matching_whatsapp_names(ContentPage: Any, WhatsAppTemplate: Any) -> None:
+    templates = WhatsAppTemplate.objects.all()
+    for template in templates:
+        pages = ContentPage.objects.filter(is_whatsapp_template=True).filter(
+            whatsapp_title=template.name
+        )
+        while pages.count() > 1:
+            page = pages.first()
+            suffix = 1
+            candidate_whatsapp_title = template.name
+            while ContentPage.objects.filter(
+                whatsapp_title=candidate_whatsapp_title
+            ).exists():
+                suffix += 1
+                candidate_whatsapp_title = f"{template.name}-{suffix}"
+            page.whatsapp_title = candidate_whatsapp_title
+            page.save(update_fields=["whatsapp_title"])
 
 
 def migrate_content_page_templates_to_standalone_templates(
-    ContentPage: Any, WhatsAppTemplate: Any
+    ContentPage: Any, WhatsAppTemplate: Any, Image: Any
 ) -> None:
     content_pages = ContentPage.objects.filter(is_whatsapp_template=True)
 
     for content_page in content_pages:
         whatsapp_block = content_page.whatsapp_body[0]
         whatsapp_value = whatsapp_block.value
+        if (
+            hasattr(whatsapp_value, "_meta")
+            and whatsapp_value._meta.verbose_name == "WhatsApp Template"
+        ):
+            continue
+        image = whatsapp_value.get("image", None)
+        if image:
+            if isinstance(image, int):
+                image = Image.objects.get(id=image)
+            elif hasattr(image, "id"):
+                image = Image.objects.get(id=image.id)
+            elif isinstance(image, dict) and "id" in image:
+                image = Image.objects.get(id=image["id"])
+            else:
+                image = None
+        example_values = list(whatsapp_value.get("example_values", []))
         whatsapp_template = WhatsAppTemplate.objects.create(
-            name=content_page.whatsapp_template_name,
+            name=content_page.whatsapp_title.lower().replace(" ", "_"),
```

Comment:
We could still have conflicts of template names because of this, `Template 1` and `template_1` won't be caught in the `rename` functions, but we'll get an integrity error here.

I think instead of trying to prep the data in the `rename` functions, we should instead create a candidate name here, and make it unique against all existing whatsapp templates, before saving it. Then we don't need to worry about where it's coming from, we just take it one template at a time, creating unique names.